skimage/future/graph/rag.py

```diff
@@ -249,6 +249,15 @@ def copy(self):
         g.max_id = self.max_id
         return g
 
+    def fresh_copy(self):
```

Comment:
This function seems redundant.

Reply:
With the new GraphViews (SubGraph, ReversedGraph, etc) you can’t assume that ``G.__class__()`` will create a new instance of the same graph type as ``G``. In fact, the call signature for ``__class__`` differs depending on whether ``G`` is a view or a base class. For v2.x you should use ``G.fresh_copy()`` to create a null graph of the correct type—ready to fill with nodes and edges.

In particular, without ``fresh_copy`` here, both ``copy`` and ``subgraph`` cause failures in your test suite.

Reply:
Maybe I should add a note in the docstring saying this is required when subclassing ``Graph`` with networkx v2 and doesn't cause problems for v1.

Reply:
@jarrodmillman yes, that'd be much appreciated. Thanks!

Reply:
Done.